ci/courtesy-push/courtesy-push.js

```diff
@@ -26,114 +30,175 @@ function formatDeps(depArr) {
-        if (name) {
-            const newDepsKey = Object.keys(newDepsDict).find(key => key.toLowerCase() === name.toLowerCase());
-            if (newDepsKey && newDepsDict[newDepsKey]) {
-                // update the version
-                depDetails[3] = newDepsDict[newDepsKey];
-                updatedDeps.push(depDetails.join('"'));
-
-                changedTasks.push(newDepsKey);
-                delete newDepsDict[newDepsKey];
-            } else {
-                updatedDeps.push(currentDep);
-                console.log(`"${currentDep}"`);
-            }
-        } else {
-            updatedDeps.push(currentDep);
+        if (!name || !basicDepsForUpdate.has(name)) {
+            index++;
+            continue;
         }
-    });
 
-    // add the new deps from the start
-    // working only for generated deps
-
-    if (Object.keys(newDepsDict).length > 0) {
-        for (let packageName in newDepsDict) {
-            // new deps should include old packages completely
-            // Example:
-            // Mseng.MS.TF.DistributedTask.Tasks.AndroidSigningV2-Node16(packageName) should include 
-            // Mseng.MS.TF.DistributedTask.Tasks.AndroidSigningV2(basePackageName)
-            const depToBeInserted = newDepsArr.find(dep => dep.includes(packageName));
-            const pushingIndex = findLastIndex(updatedDeps, basePackage => {
-                if (!basePackage) return false;
-
-                const depDetails = basePackage.split('"');
-                const name = depDetails[1];
-                return name && name.startsWith(msPrefix) && packageName.includes(name.split("_")[0])
-            });
+        newDepsArr.splice(index, 1, depsForUpdate[name].depStr);
+        index++;
 
-            if (pushingIndex !== -1) {
-                // We need to insert new package after the old one
-                updatedDeps.splice(pushingIndex + 1, 0, depToBeInserted);
-                changedTasks.push(packageName);
-            }
+        if (depsForUpdate[name].configs) {
+            depsForUpdate[name].configs.
+                sort((a,b) => a.name > b.name).
+                forEach(config => {
+                    updatedDepsObj.added.push(config.name);
+                    newDepsArr.splice(index, 0, config.depStr);
+                    index++;
+            });
         }
     }
-    // write it as a new file where currentDeps is
-    fs.writeFileSync(outputPath, updatedDeps.join('\n'));
+
+    return [newDepsArr, updatedDepsObj];
+}
+
+/**
+ * The main function for unified dependencies update
+ * The function will parse unified dependencies file and update it with new dependencies/remove unused
+ * Since the generated tasks could be used and build only with default version, if unified_deps.xml doesn't contain
+ * the default version, the specific config (e.g. Node16) will be removed from the list of dependencies
+* @param {String} pathToUnifiedDeps - path to UnifiedDependencies.xml
```

Comment:
```suggestion
 * The main function for unified dependencies update
 * The function parses unified dependencies file and updates it with new dependencies/remove unused
 * Since the generated tasks can only be used and build with default version, if unified_deps.xml doesn't contain
 * the default version, the specific config (e.g. Node16) will be removed from the list of dependencies
* @param {String} pathToUnifiedDeps - path to UnifiedDependencies.xml
```